constructor/construct.py

```diff
@@ -81,6 +81,11 @@
 for example, if `python=3.6` is included, then conda will always seek versions
 of packages compatible with Python 3.6. If this is option is not provided, it
 will be set equal to the value of `specs`.
+'''),
+
+    ('virtual_specs',          False, list, '''
+A list of virtual packages that must be satisfied at install time.Virtual
+packages must start with `__`. For example, `__osx>=11` or `__glibc>=2.24`.
```

Comment:
This isn't enforced anywhere in the code. Should the specs be sanitized somehow or an error be thrown for non-virtual packages?

Reply:
True. I was actually wondering whether we should turn this into a more abstract `solver_checks`? Although I'm not sure why that'd be useful.

Reply:
Added validation